### ISSUE TITLE
build failed for iOS, version="`defaults read "$versions_path" "$name" 2>/dev/null`" failed

### ISSUE THREAD

**Issue by ONECATYU** [MailCore/mailcore2#1886]
**Summary**
The build failed on iOS because the `defaults read` command failed to read `deps-versions.plist` causing the download dependency to fail

**Try**:
Replace `defaults read "$versions_path" "$name" 2>/dev/null` with `$(/usr/libexec/PlistBuddy -c "Print $name" $versions_path)`

**Comment by haithngn** [comment]
could you please share your Podfile at MailCore2 dependency declaration?

**Comment by ONECATYU** [comment]
I noticed that pod Install's `libmailcore-ios.a` was missing some information, so I clone the code on the Master branch and built it, but the build failed. The reason was that the `defaults read` `deps-versions.plist` failed, making it impossible to download the dependencies for the corresponding version. So I replaced the `defaults read` command with `$(/usr/libexec/PlistBuddy -c "Print $name" $versions_path)` and found that the build was successful. Therefore, I would like to raise this question to help others with the same problem.
